Attribute Analysis.ipynb

```diff
@@ -0,0 +1,634 @@
+    "    titles.\"\"\"\n",
+    "    # For text objects, we need to draw the figure first, otherwise the extents\n",
+    "    # are undefined.\n",
+    "    ax.figure.canvas.draw()\n",
+    "    items = ax.get_xticklabels() + ax.get_yticklabels() \n",
+    "    items += [ax, ax.title]\n",
+    "    bbox = Bbox.union([item.get_window_extent() for item in items])\n",
+    "\n",
+    "    return bbox.expanded(1.0 + pad, 1.0 + pad)\n",
+    "\n",
+    "def display_filepaths(filepaths, width=100, height=100):\n",
+    "    sidebyside = widgets.HBox([widgets.Image(value=open(filepath, 'rb').read(), format='png', width=width, height=height) for filepath in filepaths], layout=Layout(height='{}px'.format(height)))\n",
+    "    display(sidebyside)\n",
+    "\n",
+    "def dec_to_show(p):\n",
+    "    if p < .001:\n",
+    "        return '{:0.3e}'.format(p)\n",
+    "    else:\n",
+    "        return round(p, 3)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Dataset\n",
+    "Fill in below with dataset and file path names\n",
+    "<a id=\"dataset\"></a>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "transform_train = transforms.Compose([ \n",
+    "        transforms.ToTensor()\n",
+    "        ])\n",
+    "dataset = datasets.CoCoDataset(transform_train)\n",
+    "folder_name = 'coco_example'\n",
+    "\n",
+    "# dataset = datasets.OpenImagesDataset(transform_train)\n",
+    "# folder_name = 'openimages_supp'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "save_loc = '1_pager_gen'\n",
+    "os.system(\"rm -r results/{0}/{1}\".format(folder_name, save_loc))\n",
+    "file = folder(save_loc, folder_name)\n",
+    "first_pass = True\n",
+    "to_write = {}\n",
+    "if not os.path.exists(\"checkpoints/{}\".format(folder_name)):\n",
+    "    os.mkdir(\"checkpoints/{}\".format(folder_name))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data_folder = None\n",
+    "\n",
+    "distances = pickle.load(open(\"results/{}/3.pkl\".format(folder_name), \"rb\"))\n",
+    "sample_file = distances[0][0][0][3]\n",
+    "if not os.path.exists(sample_file):\n",
+    "    assert data_folder is not None, \"initialize data_folder with folder path of your data\"\n",
+    "    dataset.init_folder_path(data_folder)\n",
+    "    print(\"overwriting from_path() function\")\n",
+    "    dataset.from_path = dataset.from_path_prerun"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#attr_names = dataset.attribute_names\n",
+    "#num_attrs = len(attr_names)\n",
+    "\n",
+    "attr_names = [\"Dark\", \"Light\", \"Unsure\", \"Both\"]\n",
```

Comment:
please remove these lines to keep the attr_names = dataset.attribute_names stuff above